runtime/parachains/src/util.rs

```diff
@@ -41,15 +41,49 @@ pub fn make_persisted_validation_data<T: paras::Config + hrmp::Config>(
 	})
 }
 
-/// Take the active subset of a set containing all validators.
-pub fn take_active_subset<T: Clone>(active_validators: &[ValidatorIndex], set: &[T]) -> Vec<T> {
-	let subset: Vec<_> = active_validators
+/// Take an active subset of a set containing all validators.
+///
+/// First item in pair will be all items in set have indeces found in the `active` indices set (in
+/// the order of the `active` vec, the second item will contain the rest, in the original order.
+///
+/// ```ignore
+///		split_active_subset(active.into_iter().collect(), all).0 == take_active_subset(active, all)
+/// ```
+pub fn split_active_subset<T: Clone>(active: &[ValidatorIndex], all: &[T]) -> (Vec<T>, Vec<T>) {
+	let active_set: BTreeSet<_> = active.iter().cloned().collect();
+	// active result has ordering of active set.
+	let active_result = take_active_subset(active, all);
+	// inactive result preserves original ordering of `all`.
+	let inactive_result = all
 		.iter()
-		.filter_map(|i| set.get(i.0 as usize))
+		.enumerate()
+		.filter(|(i, _)| !active_set.contains(&ValidatorIndex(*i as _)))
+		.map(|(_, v)| v)
 		.cloned()
 		.collect();
 
-	if subset.len() != active_validators.len() {
+	if active_result.len() != active.len() {
+		log::warn!(
+			target: "runtime::parachains",
+			"Took active validators from set with wrong size.",
+		);
+	}
+
+	(active_result, inactive_result)
+}
+
+/// Uses `split_active_subset` and concatenates the inactive to the active vec.
+pub fn take_active_subset_and_inactive<T: Clone>(active: &[ValidatorIndex], all: &[T]) -> Vec<T> {
+	let (mut a, mut i) = split_active_subset(active, all);
+	a.append(&mut i);
+	a
```

Comment:
nit: iiuc is only done, to assure that the acitve validators occupy the indices `0..n`, correct? Would be nice to state this in the comment and/or the `fn`-name